src/components/DappBrowser/hooks/useBrowserScrollView.ts

```diff
@@ -0,0 +1,345 @@
+              gestureScale: 1.1,
+              gestureX: decision.translationX ?? 0,
+              isActive: true,
+              tabId: decision.tabInfo.tabId,
+              tabIndex: decision.tabInfo.tabIndex,
+            });
+            break;
+          }
+
+          case 'continueClose': {
+            if (!decision.tabInfo) return;
+
+            updateTabGestureState(activeTabCloseGestures, {
+              gestureScale: 1.1,
+              gestureX: decision.translationX ?? 0,
+              isActive: true,
+              tabId: decision.tabInfo.tabId,
+              tabIndex: decision.tabInfo.tabIndex,
+            });
+            break;
+          }
+
+          case 'ignore':
+            break;
+        }
+      })
+
+      .onTouchesCancelled((_, manager) => {
+        if (ENABLE_PAN_LOGS) console.log('[Pan Gesture] TOUCH CANCELLED');
+
+        if (touchInfo.value?.initialTappedTab) {
+          const { tabId, tabIndex } = touchInfo.value.initialTappedTab;
+          updateTabGestureState(activeTabCloseGestures, {
+            gestureScale: 1,
+            gestureX: 0,
+            isActive: false,
+            tabId,
+            tabIndex,
+          });
+        }
+
+        manager.fail();
+        gestureManagerState.value = 'inactive';
+        touchInfo.value = undefined;
+      })
+
+      .onTouchesUp((e, manager) => {
+        if (ENABLE_PAN_LOGS) console.log('[Pan Gesture] TOUCH UP');
+
+        const result = determineTapResult({
+          currentTouch: {
+            x: e.changedTouches[0].absoluteX,
+            y: e.changedTouches[0].absoluteY,
+          },
+          gestureState: gestureManagerState.value,
+          tabViewVisible: tabViewVisible.value,
+          touchInfo: touchInfo.value,
+        });
+
+        switch (result.type) {
+          case 'close':
+            gestureManagerState.value = 'inactive';
+            closeTab(result.tabInfo.tabId, result.tabInfo.tabIndex, undefined);
+            touchInfo.value = undefined;
+            break;
+
+          case 'select':
+            gestureManagerState.value = 'inactive';
+            toggleTabViewWorklet(result.tabInfo.tabIndex);
+            touchInfo.value = undefined;
+            break;
+        }
+
+        manager.end();
+        gestureManagerState.value = 'inactive';
+        touchInfo.value = undefined;
+      })
+
+      .onEnd((e, success) => {
+        if (ENABLE_PAN_LOGS) console.log('[Pan Gesture] ON END');
+
+        if (!touchInfo.value?.initialTappedTab) return;
+
+        if (!success) {
+          updateTabGestureState(activeTabCloseGestures, {
+            gestureScale: 1,
+            gestureX: 0,
+            isActive: false,
+            tabId: touchInfo.value.initialTappedTab.tabId,
+            tabIndex: touchInfo.value.initialTappedTab.tabIndex,
+          });
+          gestureManagerState.value = 'inactive';
+          touchInfo.value = undefined;
+          return;
+        }
```

Comment:
Need to better ensure the close gesture values always gets cleaned up after tab close interactions (if the tab isn't closed). Current logic might miss cleanup in certain rare cases, causing the tab to be temporarily stuck with its last `gestureScale` and `gestureX` values until you interact with it again. There's also a bit of redundancy between `onTouchesUp`, `onTouchesCancelled`, and `onEnd` that should be cleaned up.